File: DentOffice.Model/Pregled.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DentOffice.Model
{
    public class Pregled
    {
        public int PregledID { get; set; }
        public int TrajanjePregleda { get; set; }
        public string Napomena { get; set; }

        public int KorisnikId { get; set; }
        public Korisnik Korisnik { get; set; }
        public string DoktorIme { get; set; }

        public int TerminId { get; set; }
        public Termin Termin { get; set; }
        public string TerminImePacijenta { get; set; }
        public string TerminRazlog { get; set; }
        public string TerminZavrsen { get; set; }

        public int LijekId { get; set; }
        public Lijek Lijek { get; set; }
        public string LijekTekst { get; set; }

        public int DijagnozaId { get; set; }
        public Dijagnoza Dijagnoza { get; set; }
        public string DijagnozaTekst { get; set; }

        public string PregledIme { get; set; }

        public string UslugaDatum => (Termin != null && Termin.Usluga != null) ? Termin.Usluga.Naziv + " (" + Termin.DatumVrijeme.ToShortDateString() + ")" : "Nema označenog pregleda";  
        public DateTime? DatumTermina => Termin != null ? (DateTime?)Termin.DatumVrijeme : null;
    }
}

File: dentofficemobile/lib/models/pregled.dart
// -- pregled.dart --
import 'package:json_annotation/json_annotation.dart';

import 'dijagnoza.dart';
import 'korisnik.dart';
import 'termin.dart';
import 'lijek.dart';

part 'pregled.g.dart';

@JsonSerializable()
class Pregled {
  int pregledID;
  int trajanjePregleda;
  String napomena;
  int korisnikId;
  Korisnik korisnik;
  String doktorIme;
  int terminId;
  Termin termin;
  String terminImePacijenta;
  String terminRazlog;
  String terminZavrsen;
  int lijekId;
  Lijek lijek;
  String lijekTekst;
  int dijagnozaId;
  Dijagnoza dijagnoza;
  String dijagnozaTekst;
  String pregledIme;

  Pregled({required this.pregledID,required this.trajanjePregleda,required this.napomena,required this.korisnikId,required this.korisnik,required this.doktorIme,required this.terminId,required this.termin,required this.terminImePacijenta,required this.terminRazlog,required this.terminZavrsen,required this.lijekId,required this.lijek,required this.lijekTekst,required this.dijagnozaId,required this.dijagnoza,required this.dijagnozaTekst,required this.pregledIme,});

  factory Pregled.fromJson(Map<String, dynamic> json) => _$PregledFromJson(json);

  Map<String, dynamic> toJson() => _$PregledToJson(this);
}